### DISCUSSION TITLE
Assign name to constraints in C# API

### DISCUSSION THREAD

**Discussion by gastald88** [google/or-tools#4098] [Linear Solver questions] [answered]
**What language and solver does this apply to?**
C#

**Describe the problem you are trying to solve.**
I want to assign a name to a constraint of a Mixed Integer Linear Problem solver. I include such constraint with the `Add` method accepting a `LinearExpr` instance.

**Describe the solution you'd like**
Similarly to the Python interface, it would be very comfy and intuitive to have an overload of the `Add` method accepting two arguments: the constraint and its name.

**Additional context**
an example of the call would be 

```
foreach (int i in I)
    solver.Add(Sum((from k in K select a[i, j] * x[j]).ToArray()) == b[i], "my_constraint_{0}".FormatString(i));
```

where the `Sum`  function is the one provided in `LinearExprArrayHelper`.

Thank you in advance for the amazing library, API and support.

**Comment by lperron** [comment, marked as answer]
This is not implemented.

I recommend using the model_builder API, also in C# that supports it.

**Reply by gastald88** [reply]
thank you very much for your help, it works perfectly. 

As far as I could dig in, maybe not so easy to find usage examples in the web nor the doc though, it would be quite an asset to have it at hand.

Thanks again

**Comment by lperron** [comment]
There are examples in the samples directory, with the mb suffix.

**Reply by gastald88** [reply]
many thanks!
